Update documentation formatting

fidian opened the pull request:
This fixes spelling errors, grammatical errors, and makes the code blocks consistently indented to match the paragraphs and other indentation level.  Links were pulled out and moved to the bottom as well as standardized so they might be easier to update.  Removed commented code because there was another copy elsewhere as an example.  All code-related things are now `code` and non-code things may have backticks removed.

fidian committed:
Update documentation formatting

andrasq commented on Readme.md:
```diff
@@ -16,14 +16,15 @@ There are not a lot of frills yet, but I was able to swap out restify in a
 
-        var Restiq = require('restiq');
-        var app = Restiq.createServer();
+    var restiq = require('restiq');
```
the way `restiq` is structured, `require('restiq')` returns a class, and I capitalize class names.
`Restiq.createServer`, `Restiq.mw` etc are class methods / properties.

Edit: hmm, I guess that can be left as an internal implementation detail; users can interact
with `restiq` as a vanilla package.  Ok to leave.

fidian replied:
Yeah, that was my thought. You expect people to not use `new Restiq`, so I made it lowercase.

andrasq commented on Readme.md:
```diff
@@ -43,30 +44,18 @@ Comparison
-- [express](https://www.npmjs.org/package/express) - 7.9k/s
-- [restify](https://www.npmjs.org/package/restify) - 4.6k/s (8k/s using just the http methods)
-- [hapi](https://www.npmjs.org/package/hapi) - 0.2k/s* (1.8k/s with `setNoDelay()`)
+- restiq - 20.9k/s
```
lost the link to the npm package then

fidian replied:
I didn't think you needed the link when they are reading the source code.

12599bc I've linked to GitHub - would you prefer I link to npmjs.org instead?

andrasq replied:
well, they're reading the Readme, which could be on npmjs or in a forked repo.
The npmjs page has a link to github, but github doesn't link to npmjs, so I tend to
point to npmjs (that, and that's the version most people would install)

andrasq commented on Readme.md:
```diff
@@ -43,30 +44,18 @@ Comparison
-- [restify](https://www.npmjs.org/package/restify) - 4.6k/s (8k/s using just the http methods)
-- [hapi](https://www.npmjs.org/package/hapi) - 0.2k/s* (1.8k/s with `setNoDelay()`)
+- restiq - 20.9k/s
+- [http] - 17.6k/s
```
cute; I'm not familiar with this syntax, will `markdown` automatically match it up with the definitions at the bottom?
Will it render the same in both github and npmjs.org?

fidian replied:
Yes and yes. It's part of the original Markdown spec.

andrasq replied:
cool, I'll have to start using it myself

andrasq commented on Readme.md:
```diff
@@ -92,88 +81,84 @@ or in common to all routes.  Steps in common can be either before or after the
-        app.use(restify.queryParser());
-        app.get('/echo', function(req, res, next) {
-            res.send(200, req.params);
+    var Restiq = require('restiq');
```
lowercase here too?

fidian replied:
I have fixed this in 4cf4def

andrasq commented on Readme.md:
```diff
@@ -92,88 +81,84 @@ or in common to all routes.  Steps in common can be either before or after the
-        app.get('/echo', function(req, res, next) {
-            res.send(200, req.params);
+    var Restiq = require('restiq');
+    var app = Restiq.createServer({readImmediate: 0});
```
case

fidian replied:
I have fixed this in 4cf4def

andrasq commented on Readme.md:
```diff
@@ -261,65 +247,68 @@ Options:
+the body with `mw.readBody` if it has not been read already.
 
-### Restiq.mw.readBody( req, res, next )
+### `Restiq.mw.readBody( req, res, next )`
```
case

fidian replied:
I have fixed this in 4cf4def

andrasq commented on Readme.md:
```diff
@@ -261,65 +247,68 @@ Options:
 first time.
 
-### Restiq.mw.skipBody( req, res, next )
+### `Restiq.mw.skipBody( req, res, next )`
```
case

fidian replied:
I have fixed this in 4cf4def

andrasq commented on Readme.md:
```diff
@@ -92,88 +81,84 @@ or in common to all routes.  Steps in common can be either before or after the
 
-### Restiq( options )
-### Restiq.createServer( options )
+### `restiq( options )`
```
I'm not overly fond of the way `code` typefaces render in titles; try to convince me that this change is an improvement.  (I agree with the changes made inside the text body, this is specifically just in titles.)

fidian replied:
While I prefer the code in titles (since this is code-ish), I will happily return them to non-code bits of text. I defer to following your style. :-)

bb990a6

andrasq commented on Readme.md:
```diff
@@ -493,3 +500,12 @@ Todo
 - call versioning
+
+
+[express]: https://expressjs.com/
```
I like the links to the websites, but I don't like losing the links to the npmjs packages.  Ideas on how to do both?  (However, see note above about now having lost the [restiq] npmjs link.)

fidian committed:
Adding a link for restiq

fidian committed:
Making `Restiq` all lowercase everywhere it's code

fidian committed:
Removing backticks in headings

Converting

    ### `restiq.method()`

into

    ### restiq.method()

This also affects one H2

fidian committed:
Prefix `mw.method` with `restiq.`

Before:

    mw.readBody

After

    restiq.mw.readBody

fidian commented:
1. I added another commit that wasn't part of the code review. e49a89e changes middleware mentions and adds `restiq` in front of `mw` as in `mw.readBody` to `restiq.mw.readBody`.
2. I will return the links to npmjs.org if preferred. I feel like they get enough publicity in Google without any extra help and often I prefer to jump right to the repository instead of the intermediary. I can't think of a nice way to visibly show the links and agree we should just use one or the other. I leave it in your hands to decide and have no problem switching the links back.

andrasq commented:
yeah, it might be better to link to npmjs.  All the readmes should link to the
project home page, and for people like me who just want to install and try,
it saves having to search :-)

fidian committed:
Pointing packages back to npmjs.com

fidian commented:
I have linked all packages back to npmjs.com.

Thanks for creating this package, the discussion over the changes, and maintaining the codebase. :-)
